samples/matter/bridge/src/bridge_manager.cpp

```diff
@@ -0,0 +1,392 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bridge_manager.h"
+
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/util/generic-callbacks.h>
+#include <lib/support/Span.h>
+#include <lib/support/ZclString.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace ::chip;
+using namespace ::chip::app;
+
+BridgeManager BridgeManager::sBridgeManager;
+
+void BridgeManager::Init()
+{
+	/* The first dynamic endpoint is the last fixed endpoint + 1. */
+	mFirstDynamicEndpointId = static_cast<chip::EndpointId>(
+		static_cast<int>(emberAfEndpointFromIndex(static_cast<uint16_t>(emberAfFixedEndpointCount() - 1))) + 1);
+
+	mCurrentDynamicEndpointId = mFirstDynamicEndpointId;
+
+	/* Disable the placeholder endpoint */
+	emberAfEndpointEnableDisable(emberAfEndpointFromIndex(static_cast<uint16_t>(emberAfFixedEndpointCount() - 1)),
+				     false);
+}
+
+CHIP_ERROR BridgeManager::AddBridgedDevice(BridgedDevice::DeviceType bridgedDeviceType, char *nodeLabel)
+{
+	BridgedDevice *device;
+
+	char *label = nullptr;
+	if (nodeLabel && (strlen(nodeLabel) >= BridgedDevice::kNodeLabelSize)) {
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (nodeLabel) {
+		label = nodeLabel;
+	}
+
+	switch (bridgedDeviceType) {
+#if CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE
+	case BridgedDevice::DeviceType::OnOffLight: {
+		LOG_INF("Adding OnOff Light bridged device");
+		device = reinterpret_cast<BridgedDevice *>(Platform::New<OnOffLightDevice>(label));
```

Comment:
You should use public inheritance from `BridgedDevice` and then those unsafe casts wouldn't be needed.